src/YieldDisburser.sol

```diff
@@ -106,32 +109,30 @@ contract YieldDisburser is OwnableUpgradeable {
         uint256 _blockTime,
         uint256 _minVotingAmount,
         uint256 _minHoldingDuration,
-        uint256 _maxVotes,
         uint256 _maxPoints,
-        uint48 _minTimeBetweenClaims,
+        uint256 _cycleLength,
         uint48 _lastClaimedTimestamp,
         uint256 _lastClaimedBlockNumber,
         uint256 _precision
     ) public initializer {
         __Ownable_init(msg.sender);
-
         BREAD = Bread(breadAddress);
-        PRECISION = _precision;
-
-        projects = new address[](_projects.length);
-        for (uint256 i; i < _projects.length; ++i) {
+        uint256 projectLength = _projects.length;
+        projects = new address[](projectLength);
+        for (uint256 i; i < projectLength; ++i) {
             projects[i] = _projects[i];
         }
-
+        projectDistributions = new uint256[](projectLength);
         blockTime = _blockTime;
+        PRECISION = _precision;
         minVotingAmount = _minVotingAmount;
-        minHoldingDuration = _minHoldingDuration * 1 days;
-        minRequiredVotingPower = ((minVotingAmount * minHoldingDuration) * PRECISION) / blockTime;
-        maxVotes = _maxVotes;
+        minHoldingDuration = _minHoldingDuration * 1 days; // must hold for atleast _minVotingHoldingDuration  days
+        minRequiredVotingPower = ((minVotingAmount * minHoldingDuration) * PRECISION) / blockTime; // Holding minVotingAmount bread for minVotingHoldingDuration days , assuming a blockTime second block time
         maxPoints = _maxPoints;
-        minTimeBetweenClaims = _minTimeBetweenClaims * 1 days;
+        cycleLength = (_cycleLength * 1 days) / blockTime;
```

Comment:
It's not clear here what unit "cycleLength" is. it's being passed in to the constructor as "days" and then converted here into "blocks". Then on line 155 it's being added to a time stamp...

Reply:
fixed by 9e6a03a